src/clib/pio_darray.c

```diff
@@ -2898,7 +2898,40 @@ static int PIOc_read_darray_adios(file_desc_t *file, int fndims, io_desc_t *iode
         int64_t *decomp_int64_t = NULL;
         start_block_found = false;
 
-        for (size_t block = 0; block < decomp_blocks_size; block++)
+        iosystem_desc_t *ios = file->iosystem; /* Pointer to io system information. */
+        assert(ios != NULL);
+
+        /* Determine the maximum number of compute tasks assigned to each block. Round up the result of integer division. */
+        assert(decomp_blocks_size > 0);
+        int max_comptasks_per_block = ios->num_comptasks / decomp_blocks_size;
+        if (ios->num_comptasks % decomp_blocks_size != 0)
+            max_comptasks_per_block++;
+        assert(max_comptasks_per_block >= 1);
+
+        /* Using the outcomes of certain test cases, we can deduce the most probable target block we try to find.
+
+           [Case 1]
+           comp_rank = 0, num_comptasks = 4, decomp_blocks_size = 2, max_comptasks_per_block = 2, target block found = 0
+           comp_rank = 1, num_comptasks = 4, decomp_blocks_size = 2, max_comptasks_per_block = 2, target block found = 0
+           comp_rank = 2, num_comptasks = 4, decomp_blocks_size = 2, max_comptasks_per_block = 2, target block found = 1
+           comp_rank = 3, num_comptasks = 4, decomp_blocks_size = 2, max_comptasks_per_block = 2, target block found = 1
+
+           [Case 2]
+           comp_rank = 0, num_comptasks = 7, decomp_blocks_size = 3, max_comptasks_per_block = 3, target block found = 0
+           comp_rank = 1, num_comptasks = 7, decomp_blocks_size = 3, max_comptasks_per_block = 3, target block found = 0
+           comp_rank = 2, num_comptasks = 7, decomp_blocks_size = 3, max_comptasks_per_block = 3, target block found = 0
+           comp_rank = 3, num_comptasks = 7, decomp_blocks_size = 3, max_comptasks_per_block = 3, target block found = 1
+           comp_rank = 4, num_comptasks = 7, decomp_blocks_size = 3, max_comptasks_per_block = 3, target block found = 1
+           comp_rank = 5, num_comptasks = 7, decomp_blocks_size = 3, max_comptasks_per_block = 3, target block found = 1
+           comp_rank = 6, num_comptasks = 7, decomp_blocks_size = 3, max_comptasks_per_block = 3, target block found = 2
+         */
+        int guess_block = ios->comp_rank / max_comptasks_per_block;
+        assert(guess_block >= 0 && guess_block < decomp_blocks_size);
+
+        /* Since we are uncertain if our guessed target block is accurate for all scenarios, iterate through the blocks
+         * from guess_block to the end, and then from 0 to guess_block - 1. */
+        size_t block = guess_block;
+        while (true)
```

Comment:
Can we replace while(true) by while(!start_block_found) ?

Reply:
OK, we can make this change.